pkg/agent/openflow/network_policy.go

```diff
@@ -1012,3 +1046,25 @@ func (c *client) DeletePolicyRuleAddress(ruleID uint32, addrType types.AddressTy
 	// Update the Openflow entries on the OVS bridge, and update local cache.
 	return c.applyConjunctiveMatchFlows(changes)
 }
+
+func (c *client) GetNetworkPolicyFlowKeys(npName, npNamespace string) []string {
+	flowKeys := []string{}
+	// Hold replayMutex write lock to protect flows from being modified by
+	// NetworkPolicy updates and replayPolicyFlows. This is more for logic
+	// cleanliness, as: for now flow updates do not impact the matching string
+	// generation; NetworkPolicy updates do not change policyRuleConjunction.actionFlows;
+	// and last for protection of clause flows, conjMatchFlowLock is good enough.
+	c.replayMutex.Lock()
+	defer c.replayMutex.Unlock()
+
+	c.policyCache.Range(func(key, value interface{}) bool {
+		conj := value.(*policyRuleConjunction)
+		if conj.npName == npName && conj.npNamespace == npNamespace {
+			// There can be duplicated flows added due to conjunctive matches
+			// shared by multiple policy rules (clauses).
+			flowKeys = append(flowKeys, conj.getAllFlowKeys()...)
+		}
+		return true
+	})
+	return flowKeys
```

Comment:
For each conjunction, the flows are output in the order of action flow ->conjunctive match flow -> drop flows. But if there are multiple conjunction in one Network Policy Rule, the flows are mixed together. Do we have some plans to group the results according to conjunction?

Reply:
Sorry, do you mean you want to put action flows and excepts flows of all conjunctions before conjunctive match flows?

Reply:
And do you mean NetworkPolicy rule that ends up with multiple ofRules (as one ofRule has only one conjunction now)?